state/status.go

```diff
@@ -101,6 +101,10 @@ type setStatusParams struct {
 // setStatus inteprets the supplied params as documented on the type.
 func setStatus(st *State, params setStatusParams) (err error) {
 	defer errors.DeferredAnnotatef(&err, "cannot set status")
+	if params.updated == nil {
```

Comment:
I think if there was no way to generate a reasonable value, then yes, it should return an error. But since it has access to a clock through the state object, using that value is better than returning an error.